src/appleseed/renderer/modeling/postprocessingeffect/vignettepostprocessingeffect.h

```diff
@@ -0,0 +1,87 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2020 Tiago Chaves, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+// appleseed.renderer headers.
+#include "renderer/modeling/postprocessingeffect/ipostprocessingeffect.h"
+
+// appleseed.foundation headers.
+#include "foundation/math/vector.h"
+
+// Standard headers.
+#include <cstddef>
+
+// Forward declarations.
+namespace renderer  { class Frame; }
+
+namespace renderer
+{
+
+//
+// Vignette-specific parameters.
+//
+
+struct VignetteParams {
```

Comment:
In other places, we generally don't create a struct for the params. We simply give params and we do `params.get_float(...)" or something where we need the actual values. Is there a specific reason you created this struct ?

Reply:
Do you mean using a `ParamArray`? I had originally thought about it, but as Franz mentioned on Discord, they shouldn't be used to pass values "internally", for this we can use standard arguments.

Thus, I preferred to group them into a POD-struct as I think it's clean and easier to spot them this way, but I can rewrite `VignetteApplierFactory` to receive each one separately. What do you think?

Reply:
I misunderstood the code, my bad :) You did right